opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexCheck.java

```diff
@@ -38,89 +38,113 @@
+    public static boolean check(List<String> subFilesList) {
         RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         File indexRoot = new File(env.getDataRootPath(), IndexDatabase.INDEX_DIR);
         LOGGER.log(Level.FINE, "Checking for Lucene index version mismatch in {0}",
                 indexRoot);
+        int ret = 0;
 
         if (!subFilesList.isEmpty()) {
             // Assumes projects are enabled.
             for (String projectName : subFilesList) {
                 LOGGER.log(Level.FINER,
                         "Checking Lucene index version in project {0}",
                         projectName);
-                checkDir(new File(indexRoot, projectName));
+                ret |= checkDirNoExceptions(new File(indexRoot, projectName));
             }
         } else {
             if (env.isProjectsEnabled()) {
                 for (String projectName : env.getProjects().keySet()) {
                     LOGGER.log(Level.FINER,
                             "Checking Lucene index version in project {0}",
                             projectName);
-                    checkDir(new File(indexRoot, projectName));
+                    ret |= checkDirNoExceptions(new File(indexRoot, projectName));
                 }
             } else {
                 LOGGER.log(Level.FINER, "Checking Lucene index version in {0}",
                         indexRoot);
-                checkDir(indexRoot);
+                ret |= checkDirNoExceptions(indexRoot);
             }
         }
+
+        return ret == 0;
+    }
+
+    private static int checkDirNoExceptions(File dir) {
+        try {
+            checkDir(dir);
+        } catch (Exception e) {
+            LOGGER.log(Level.WARNING, "Index check for directory " + dir.toString() + " failed", e);
+            return 1;
+        }
+
+        return 0;
     }
 
     /**
-     * Check index version in given directory. It assumes that that all commits
+     * Check index in given directory. It assumes that that all commits (if any)
      * in the Lucene segment file were done with the same version.
      *
      * @param dir directory with index
-     * @thows IOException if the directory cannot be opened
+     * @throws IOException if the directory cannot be opened
+     * @throws IndexVersionException if the version of the index does not match Lucene index version
      */
-    private static void checkDir(File dir) throws Exception {
-        LockFactory lockfact = NativeFSLockFactory.INSTANCE;
+    public static void checkDir(File dir) throws IndexVersionException, IOException {
+        LockFactory lockFactory = NativeFSLockFactory.INSTANCE;
         int segVersion;
 
-        try (Directory indexDirectory = FSDirectory.open(dir.toPath(), lockfact)) {
-           SegmentInfos segInfos = null;
+        try (Directory indexDirectory = FSDirectory.open(dir.toPath(), lockFactory)) {
+           SegmentInfos segInfos;
```

Comment:
No need for separate declaration:
`SegmentInfos segInfos = SegmentInfos.readLatestCommit(indexDirectory);`